src/components/log/Log.js

```diff
@@ -2,20 +2,58 @@ import React, { Component } from "react";
 import "./Log.css";
 
 class Log extends Component {
-  constructor(props){
+  constructor(props) {
     super(props);
+
+    this.logLevels = [
+      { id: "error", name: "error" },
+      { id: "warn", name: "warn" },
+      { id: "info", name: "info" },
+      { id: "debug", name: "debug" },
+      { id: "spam", name: "spam" },
+    ];
+    this.state = { currentLoglevel: "info" };
   }
+
+  componentDidMount() {
+    let params = new URLSearchParams(window.location.search);
+    let logLevel = params.get("log");
+
+    if (this.logLevels.find((level) => level.id === logLevel)) {
+      this.setState({ currentLoglevel: logLevel });
+      this.props.callback(logLevel);
+    }
+  }
+
+  handleLevelChange(event) {
+    this.setState({ currentLoglevel: event.target.value });
+    this.setUrlParams(event.target.value);
+    this.props.callback(event.target.value);
```

Comment:
Is this callback prop guaranteed to be present?

Reply:
Yes, it's given to the Log component in app.js